### PR TITLE
Correctly and explicitly report too-short files

### LINKED ISSUE #2508: Incorrect exception thrown while passing a non-sst file to lakeFS's SST parser
The parser fails with IlligalArgumentException on getting a non-sst file instead of failing with BadFileFormatException in this case. 

After fixing this bug, we need to enable the unit test described as "with non-sst file of size 0" which is currently ignored. This unit-test is implemented under BlockParserSpec.scala.

### PR DESCRIPTION
**Pull request by arielshaqed**
Fixes #2508.

### CODE REVIEW

**Comment by arielshaqed** [comment]
Proposed changelog (in _Spark client_, **not** the lakeFS release):
> Report the correct error when a file is too short.

File: clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala [hunk 1]
```diff
@@ -392,38 +414,42 @@ class GolangContainerSpec extends AnyFunSpec with ForAllTestContainer {
...
 
-  /**
-   * Copies data from a file inside the test container to a temporary file.
-   * @param baseFileName the file name without a suffix of the files to copy from within the container
-   * @param suffix of the file to copy from the container
-   * @return
+  /** Copies data from a file inside the test container to a temporary file.
+   *  @param baseFileName the file name without a suffix of the files to copy from within the container
+   *  @param suffix of the file to copy from the container
+   *  @return
```
Comment (talSofer) [inline clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala:427]:
```suggestion
  /**
   *  Copies data from a file inside the test container to a temporary file.
   *  @param baseFileName the file name without a suffix of the files to copy from within the container
   *  @param suffix of the file to copy from the container
   *  @return
```

File: clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala [hunk 2]
```diff
@@ -392,38 +414,42 @@ class GolangContainerSpec extends AnyFunSpec with ForAllTestContainer {
...
+    )
 
-  /**
-   * Lightweight fixture for running particular tests with SSTables and JSON fules generated by a go application on
-   * test startup. The fixture uses sstables as the parser input, and the json as the source of the expected output of
-   * the parsing operation.
+  /** Lightweight fixture for running particular tests with SSTables and JSON fules generated by a go application on
+   *  test startup. The fixture uses sstables as the parser input, and the json as the source of the expected output of
+   *  the parsing operation.
    */
```
Comment (talSofer) [inline clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala:448]:
```suggestion
  /** 
   *  Lightweight fixture for running particular tests with SSTables and JSON fules generated by a go application on
   *  test startup. The fixture uses sstables as the parser input, and the json as the source of the expected output of
   *  the parsing operation.
   */
```

**Review by talSofer — APPROVED** [review]
Looks good!

File: clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala (hunk 1 above)
Comment (arielshaqed) [reply clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala:427]: Not done.  We actually *enforce* `scalafmtCheck`, so I ran `sbt scalafmt`.  I refuse to have anything further to do with formatting once the automated tool has done its thing.

Consider opening a PR to fix whatever `scalafmt` options are causing this, or to run just the one `scalafmtCheck` (I think that it runs it twice, once for each Scala version), or anything else.  Just out of scope for this PR.

File: clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala (hunk 2 above)
Comment (arielshaqed) [reply clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala:448]: As above, out of scope.

**Review by arielshaqed — COMMENTED** [review]
Thanks!

**Pull request merged**